physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,915 @@
+      ntimes = ntimes + 1
+   enddo
+   if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'land_iau_init: ntimes = ',ntimes
+   Land_IAU_Control%ntimes = ntimes
+   if (ntimes < 1) then
+      print*, "Error! in Land IAU init: ntimes < 1"
+      errmsg = "Error! in Land IAU init: ntimes < 1"
+      errflg = 1
+      ! Land_IAU_Control%do_land_iau=.false.
+      return
+   endif
+   if (ntimes > 1) then
+      allocate(idt(ntimes-1))
+      idt = Land_IAU_Control%iaufhrs(2:ntimes)-Land_IAU_Control%iaufhrs(1:ntimes-1)
+      do k=1,ntimes-1
+         if (idt(k) .ne. Land_IAU_Control%iaufhrs(2)-Land_IAU_Control%iaufhrs(1)) then
+            print *,'in land_iau_init: forecast intervals in iaufhrs must be constant'
+         !   call mpp_error (FATAL,' forecast intervals in iaufhrs must be constant')
+            errmsg = 'Fatal error in land_iau_init. forecast intervals in iaufhrs must be constant'
+            errflg = 1
+            return
+         endif
+      enddo
+      deallocate(idt)
+   endif
+   dt = (Land_IAU_Control%iau_delthrs*3600.)
+   Land_IAU_Data%rdt = 1.0/dt   !rdt
+   if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'land_iau interval, rdt',Land_IAU_Control%iau_delthrs,Land_IAU_Data%rdt
+
+   ! Read all increment files at iau init time (at beginning of cycle) 
+   ! allocate (wk3_stc(n_t, 1:im,jbeg:jend, 1:km))  
+   call read_iau_forcing_fv3(Land_IAU_Control, Land_IAU_state%stc_inc, Land_IAU_state%slc_inc, errmsg, errflg)  !, wk3_stc, wk3_slc
+   ! call read_iau_forcing_fv3(Land_IAU_Control, Land_IAU_state%inc1%stc_inc, Land_IAU_state%inc1%slc_inc, errmsg, errflg)
+   if (errflg .ne. 0) return
+   
+   ! increments already in the fv3 grid--no need for interpolation       
+   ! do k = 1, npz  ! do k = 1,n_soill    !  
+   !    do j = 1, nlat
+   !       do i = 1, nlon                
+   !          Land_IAU_state%inc1%stc_inc(i,j,k)  = wk3_stc(1, i, j, k)
+   !          Land_IAU_state%inc1%slc_inc(i,j,k)  = wk3_slc(1, i, j, k) 
+   !       end do
+   !     enddo
+   ! enddo
+
+   if (ntimes.EQ.1) then  ! only need to get incrments once since constant forcing over window
+      call setiauforcing(Land_IAU_Control, Land_IAU_Data, Land_IAU_state)
+      Land_IAU_Data%itnext = 0
+   endif
+   if (ntimes.GT.1) then  !have multiple files, but only need 2 at a time and interpoalte for timesteps between them  
+       ! interpolation is now done in land_iau_mod_getiauforcing 
+      Land_IAU_Data%hr2=Land_IAU_Control%iaufhrs(2)   
+      Land_IAU_Data%itnext = 2
+      ! ! Land_IAU_Data%snow_land_mask(:, :)  = wk3_slmsk(1, :, :)
+      ! do k = 1, npz  ! do k = 1,n_soill    !  
+      !    do j = 1, nlat
+      !       do i = 1, nlon                
+      !          Land_IAU_state%inc2%stc_inc(i,j,k)  = wk3_stc(2, i, j, k)
+      !          Land_IAU_state%inc2%slc_inc(i,j,k)  = wk3_slc(2, i, j, k) 
+      !       end do
+      !     enddo
+      ! enddo
+   endif
+!   print*,'end of IAU init',dt,rdt

```

Comment:
remove all these interpolation related commented out lines?

Reply:
Or maybe just have a comment that says 'code assumes increments are on model native grid'